Added a simple test suite using `mocha`

NikkelM opened the pull request:
I recently migrated my existing project to use webpack and found your boilerplate quite useful for this transition. However, I ran into some unexpected problems when I tried to get my test suite running with this new environment, which turned out to be mostly because of a missing `"type": "module"` in the `package.json`, which I found is not very clear when first looking at the problem.
To maybe prevent others from also falling into this trap, I added some very basic boilerplate for tests, so feel free to add this to the existing template if you find it to be useful.

sszczep commented:
Hey @NikkelM, sorry for such a late reply. 

Thanks for your PR. LGTM, will double check it in upcoming days and probably merge.

sszczep commented on .eslintrc:
```diff
@@ -11,7 +11,8 @@
     "browser": true,
-    "node": true
+    "node": true,
+	"mocha": true
```
Wrong indentation level

NikkelM replied:
Looks like this was due to tabs vs. spaces as well

sszczep commented on test/setup.js:
Use 2 space indentation instead of tabs

sszczep commented on .mocharc.json:
Use 2 space indentation instead of tabs

sszczep commented on test/storage.test.js:
Use 2 space indentation instead of tabs

sszczep reviewed (changes requested):
I left some comments. Nothing serious, just some minor indentation issues. 

I will merge once the issues are resolved. Let me know if you do not have time to do this, I will happily fix that for you.

NikkelM commented:
Thanks @sszczep for the review and noticing the issues with indentation, I've fixed those.
I also thought it might be nice to provide a simple Github Actions workflow to run the tests when pushing to `main` or opening a PR against `main`, so I've added a `.yml` for that - if you think this goes too far and want to keep the boilerplate smaller, we can remove that again. You can see how that workflow looks and runs [here](https://github.com/NikkelM/chrome-extension-webpack-tests/actions/runs/5163007303) in my fork.

sszczep commented:
Looks good to me. I think it is a great addition to that project. From what it seems the workflow misses dependencies caching, however, it is not a problem for now. All of those things were on my todo-list but I had to abandon that project for some time. I am willing to release a new version introducing CRA-like tooling:

- multiple frontend frameworks support (Vue/Angular/React/Preact/Svelte),
- Chromium/Firefox/Safari(?) support,
- optional TypeScript support, 

unfortunately, I couldn't find time for that yet.

Once again thank you for your contribution and keeping the project alive 🎇
